scripts/common

```diff
@@ -3,27 +3,50 @@ set -e
 # The scala/scala checkout directory (set by Jenkins, or `/home/travis/build/scala/scala` on travis)
 WORKSPACE="${WORKSPACE-`pwd`}"
 
+function generateAddJCenterCachePlugin() {
+  if [[ $(uname -s) == CYGWIN* ]]; then
+    homeDir=$(cygpath -u "$USERPROFILE")
+  else
+    homeDir=$HOME
+  fi
+
+  mkdir -p "$homeDir/.sbt/0.13/plugins"
+  cat > "$homeDir/.sbt/0.13/plugins/AddJCenterCache.scala" << EOF
+import sbt._
+object AddJCenterCache {
+  def apply(rs: Seq[Resolver]): Seq[Resolver] = {
+    val jcc = "jcenter-cache" at "https://scala-ci.typesafe.com/artifactory/jcenter/"
+    val i = rs.indexWhere({
+      case MavenRepository(_, r) if r.startsWith("https://repo1.maven.org/maven2") => true
+      case _ => false
+    })
+    assert(i != -1, rs)
+    rs.patch(i, List(jcc), 0)
+  }
+}
+EOF
+}
+
 # On Jenkins, each job needs its own ivy2 cache to avoid conflicts between jobs. On travis, it's ~/.ivy2.
 if [ "$TRAVIS" = "true" ]; then
   IVY2_DIR="$HOME/.ivy2"
+  sbtArgs=()
 else
   IVY2_DIR="$WORKSPACE/.ivy2"
+  generateAddJCenterCachePlugin
+  addJCenterCache="set externalResolvers := AddJCenterCache(externalResolvers.value)"
```

Comment:
TODO: this only adds the resolver to the root (sbt) project. Using `set externalResolvers in Global := AddJCenterCache((externalResolvers in Global).value)` doesn't work, it gives `Reference to undefined setting`. Still trying to figure this out.